cmake/std/PullRequestLinuxDriverTest.py

```diff
@@ -42,706 +40,221 @@
-              'The error code was: {}'.format(cpe.returncode))
-
-    return None
-
-
-def validateSourceBranchIfDestBranchIsMaster(arguments):
-    """
-    We only allow special branches to be merged into master.
-    These should be protected branches that not everyone can create.
-    This routine checks that the source branch name is matches this form:
-    -  master_merge_YYYYMMDD_HHMMSS
+    # Banner
+    print("+" + "="*78 + "+")
+    print("|")
+    print("|   T R I L I N O S   P U L L R E Q U E S T   D R I V E R   S T A R T")
+    print("|")
+    if args.dry_run:
+        print("|                D R Y   R U N   M O D E   E N A B L E D")
+        print("|")
+    print("+" + "="*78 + "+")
+    print("")
 
-    Script will exit with nonzero exit code if this test fails.
-    """
-    re_src_branchname = "master_merge_[0-9]{8}_[0-9]{6}"
-    if "master" == arguments.targetBranch:
-        if re.match(re_src_branchname, arguments.sourceBranch) is None:
-            # If destination is master and source isn't master_merge_YYYYMMDD_HHMMSS
-            # then we exit here with nonzero exit status.
-            sys.exit(dedent("""\
-            ------------------------------------------------------------------------------------------
-            NOTICE: Destination branch is trilinos/Trilnos::master
-            ERROR : Source branch is NOT trilinos/Trilinos::master_merge_YYYYMMDD_HHMMSS
-                  : This violates Trilinos policy, pull requests into the master branch are restricted.
-                  : Perhaps you forgot to specify the develop branch as the target in your PR?
-            ------------------------------------------------------------------------------------------
-            """))
-        else:
-            print(dedent("""\
-                NOTICE: Source branch IS trilinos/Trilinos::{sourceBranch}
-                        : This is allowed, proceeding with testing.""".format(**vars(arguments))
-                        )
-                 )
+    if 'standard' == args.mode:
+        pr_config = trilinosprhelpers.TrilinosPRConfigurationStandard(args)
+    elif 'installation' == args.mode:
+        pr_config = trilinosprhelpers.TrilinosPRConfigurationInstallation(args)
     else:
-        print(dedent("""\
-            NOTICE: Destination branch is NOT trilinos/Trilinos::master"
-                  : PR testing will proceed."""))
+        raise KeyError("ERROR: Unknown test mode, {}, was provided.".format(args.mode))
 
+    pr_config.prepare_test()
+    pr_config.execute_test()
     return 0
 
 
-def generateBuildNameString(arguments):
-    """
-    Generate the build name string
-    PR-<PR Number>-test-<Jenkins Job Name>-<Jenkins Build Number>
-    """
-    output = "PR-{github_pr_number}-test-{job_base_name}-{job_number}".format(**vars(arguments))
-    return output
-
 
+if __name__ == "__main__":
+    args = parse_args()
+    status = main(args)
+    print("Done.")
+    sys.exit(status)
```

Comment:
Do you need anything for catching/handling exceptions?  You may not, but we usually have a `try`/`catch` block in `main`.

Reply:
I don't think it really adds any value in this particular spot...  There are places where things woof and I am cool with the traceback just ending the application at that spot.  Generally that means something pretty bad went wrong and I don't really need another layer of traceback since I'll still have it anyhow. All a try...catch in this spot would be useful for is as a pass-through for an error. /shrug